src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/GeneratedComInterfaceAnalyzerTests.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.Interop;
+using Microsoft.Interop.UnitTests;
+using Xunit;
+
+namespace ComInterfaceGenerator.Unit.Tests
+{
+    public class GeneratedComInterfaceAnalyzerTests
+    {
+        static string _usings = $$"""
+            #pragma warning disable CS8019
+            using System.Runtime.InteropServices.Marshalling;
+            using System.Runtime.InteropServices;
+            #pragma warning restore CS8019
+            """;
+
+        public static async Task<ImmutableArray<Diagnostic>> CompileAndRunAnalyzer(params string[] codeSnippets)
+        {
+            Compilation comp = await TestUtils.CreateCompilation(codeSnippets);
+            // Ensure no compiler errors
+            Assert.Empty(comp.GetDiagnostics());
+            // Ensure no analyzer erros
+            return await TestUtils.RunAnalyzers(comp, new GeneratedComInterfaceAttributeAnalyzer());
+        }
```

Comment:
Instead of creating your own compilation and running the analyzers, we should use the Roslyn Analyzer Test SDK here as we do for the ConvertToLibraryImportAnalyzerTests test suite. That test will handle validating all of the diagnostic locations and ids at a much higher fidelity than just checking the number of diagnostics.